packages/workspace/src/core/project-graph/build-dependencies/explicit-npm-dependencies.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  AddProjectDependency,
+  DependencyType,
+  ProjectGraphContext,
+  ProjectGraphNodeRecords
+} from '../project-graph-models';
+import { TypeScriptImportLocator } from './typescript-import-locator';
+
+export function buildExplicitNpmDependencies(
+  ctx: ProjectGraphContext,
+  nodes: ProjectGraphNodeRecords,
+  addDependency: AddProjectDependency,
+  fileRead: (s: string) => string
+) {
+  const importLocator = new TypeScriptImportLocator(fileRead);
+
+  Object.keys(ctx.fileMap).forEach(source => {
+    Object.values(ctx.fileMap[source]).forEach(f => {
+      importLocator.fromFile(
+        f.file,
+        (importExpr: string, filePath: string, type: DependencyType) => {
+          const target = nodes[importExpr];
```

Comment:
Someone can import from `@angular/core/testing` which would not appear in `package.json`.

Can this logic be in the existing explicit npm dependency locator?